lib/src/mock_query.dart

```diff
@@ -323,21 +323,40 @@ class MockQuery<T extends Object?> extends FakeQueryWithParent<T> {
   }
 
   @override
-  Query<T> endAtDocument(DocumentSnapshot documentSnapshot) {
-    // TODO: implement endAtDocument
-    throw UnimplementedError();
+  Query<T> endAtDocument(DocumentSnapshot snapshot) {
+    return MockQuery(this, (docs) {
+      final index = docs.indexWhere((doc) {
+        return doc.id == snapshot.id;
+      });
+
+      return docs.sublist(0, index == -1 ? docs.length : index + 1);
+    });
   }
 
   @override
-  Query<T> endBefore(List values) {
-    // TODO: implement endBefore
-    throw UnimplementedError();
-  }
+  Query<T> endBefore(List values) => _cursorUtil(
+      orderByKeys: parameters['orderedBy'] ?? [],
+      values: values,
+      f: (docs, index, exactMatch) => docs.sublist(
+            0,
+            index == -1 ? docs.length - 1 : index,
+          ));
 
   @override
-  Query<T> endBeforeDocument(DocumentSnapshot documentSnapshot) {
-    // TODO: implement endBeforeDocument
-    throw UnimplementedError();
+  Query<T> endBeforeDocument(DocumentSnapshot snapshot) {
+    return MockQuery(this, (docs) {
+      final index = docs.indexWhere((doc) {
+        return doc.id == snapshot.id;
+      });
+
+      if (index == -1) {
+        throw PlatformException(
```

Comment:
How come in the other queries, when it's not found, it returns everything. But for this one query, it throws an exception? Is that the behavior in Firestore?

Reply:
I copied the code from ```startAfterDocument``` and tweaked it so that they have the same logic, but I honestly don't know how firestore behaves in this case

Reply:
Alright. It's ok, we can check the actual behavior and refine later.